common/stream.cpp

```diff
@@ -28,6 +28,14 @@
 
 namespace Common {
 
+void WriteStream::writeStream(ReadStream *stream, uint32 dataSize) {
+	void *buf = malloc(dataSize);
```

Comment:
Typically data copying routines use some kind of buffer of a fixed size as temporary storage so that they can work on arbitrarily sized inputs. See what boost does for example:
https://github.com/boostorg/iostreams/blob/develop/include/boost/iostreams/copy.hpp#L123

Reply:
This is indeed nice to have, but I believe that it's outside of the scope of this pull request. The code can be adapted later on in-tree